nuvla/job_engine/connector/kubernetes.py

```diff
@@ -173,46 +197,86 @@ def start(self, **kwargs):
         env = kwargs['env']
         files = kwargs['files']
         manifest = kwargs['docker_compose']
-        namespace = kwargs['name']
+        # This is the deployment ID.
+        deployment_uuid = kwargs['name']
         registries_auth = kwargs['registries_auth']
 
+        custom_namespaces = Kubernetes._get_all_namespaces_from_manifest(manifest)
+        log.debug('Namespaces from manifest: %s', custom_namespaces)
+        if len(custom_namespaces) > 1:
```

Comment:
Minor suggestion. But can you remind me why we are adding this constraint about single namespace allowed in manifest? Since we can support many with your current code state?
 
```
ns_len = len(custom_namespaces)
if ns_len > 1:
  msg = (f'Only single namespace allowed in manifest. Found:'
              f' {custom_namespaces}')
  log.error(msg)
elif ns_len == 1:
  namespace = list(custom_namespaces)[0]
else:
  namespace = deployment_uuid
```

Reply:
I simply wanted to stick to the definition of the task where the requirement was to deploy to a single namespace. Also, I didn't have enough time to think about and test probable corner cases when multiple namespaces are defined by the user. The Epic will not be closed so this will be worked on later, and we will be able to remove this restriction.